File: lingting-spring-jackson/src/test/java/live/lingting/spring/jackson/TestObjectMapperAfter.java
package live.lingting.spring.jackson;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.stereotype.Component;

import static org.junit.jupiter.api.Assertions.assertFalse;

/**
 * @author lingting 2024-02-02 16:00
 */
@Component
public class TestObjectMapperAfter implements ObjectMapperAfter {

	private boolean after = false;

	@Override
	public void apply(ObjectMapper mapper) {
		assertFalse(after);
		after = true;
	}

	public boolean isAfter() {
		return this.after;
	}

}

File: lingting-spring-jackson/src/test/kotlin/live/lingting/spring/jackson/TestObjectMapperAfter.kt
package live.lingting.spring.jackson

import com.fasterxml.jackson.databind.ObjectMapper
import org.junit.jupiter.api.Assertions
import org.springframework.stereotype.Component

/**
 * @author lingting 2024-02-02 16:00
 */
@Component
class TestObjectMapperAfter : ObjectMapperAfter {
    var isAfter: Boolean = false
        private set

    override fun apply(mapper: ObjectMapper) {
        Assertions.assertFalse(this.isAfter)
        this.isAfter = true
    }
}
